src/App.js

```diff
@@ -14,11 +15,74 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: goodsFromServer,
+    listVisebility: false,
+  }
+
+  showGoods = (event) => {
+    this.setState({
+      listVisebility: true,
+    });
+    const startButton = event;
+
+    startButton.target.style.visibility = 'hidden';
+  }
+
+  reverseGoods = () => {
+    this.setState(state => ({
+      goods: [...state.goods].reverse(),
```

Comment:
```suggestion
      goods: [...state.goods] 
```
Лучше один раз сделать копию в переменную.

Reply:
Мені ж всерівно потрібно буде кожен раз робити копію при внесенні змін в state, я просто не розумію навіщо копію тут переносити в окрему перемінну. Якщо б я міняв масив в самому рендері, то, напевно так, було б логічно використати перемінну

Reply:
можно не хранить массив в state, хранить в state только то что надо делать  с массивом isReversed,
    sortBy. В рендере GoodsList уже преобразовывать.

Reply:
Ты прав. Можно не хранить goods в state, хранить только то как его менять isReversed, sortBy, и в рендере делать одну копию.